File: BackEnd/src/domain/interfaces/usecases/chat/IGetConversationsUseCase.ts
/**
 * @file IGetConversationsUseCase.ts
 * @description Interfaz del caso de uso para obtener las conversaciones del usuario.
 * @module Chat
 * @layer Domain
 */

import type { ConversationResponseDTO } from '../../../dtos/chat/conversation-response.dto.js';

/**
 * Contrato del use case GetUserConversations.
 * Devuelve todas las conversaciones activas del usuario, enriquecidas con
 * datos del otro participante, último mensaje y contador de no leídos.
 */
export interface IGetConversationsUseCase {
  /**
   * Ejecuta el caso de uso.
   *
   * @param userId — ID del usuario autenticado.
   * @returns Lista de ConversationResponseDTOs ordenada por actividad reciente.
   */
  execute(userId: string): Promise<ConversationResponseDTO[]>;
}

File: FrontEnd/lib/domain/interfaces/usecases/chat/i_get_conversations_use_case.dart
/// @file i_get_conversations_use_case.dart
/// @description Interfaz: Obtiene las conversaciones del usuario autenticado.
/// @module Chat
/// @layer Domain
library;

import '../../../entities/conversation.dart';

// ═══════════════════════════════════════════════════════════════════════════════
// I GET CONVERSATIONS USE CASE
// ═══════════════════════════════════════════════════════════════════════════════

/// Contrato del use case que recupera la lista de conversaciones activas.
abstract interface class IGetConversationsUseCase {
  /// Devuelve las conversaciones activas del usuario, ordenadas por último mensaje.
  ///
  /// [param] forceRefresh — si true, salta la caché local (TTL 1 min) y
  ///         consulta directamente la API. Usado al recibir `message:received`
  ///         por socket para reflejar el nuevo unreadCount al instante.
  /// [returns] lista de [Conversation].
  /// [throws] AppError — si la carga falla.
  Future<List<Conversation>> execute({bool forceRefresh = false});
}
